src/state/branch.rs

```diff
@@ -85,7 +74,60 @@ impl Branch {
         self.branches.height() == 0
     }
 
-    /// Returns the new node's id in the branch
+    /// Returns the node id of the best tip
+    pub fn best_tip_id(&self) -> NodeId {
```

Comment:
did this already exist on the branch class?

Reply:
Wow vestigial code from the past that was present in `snapshot` and got included in the cherry-pick! Removed.